osbrain/tests/test_proxy.py

```diff
@@ -119,6 +122,75 @@ def ping(self):
         InitTimeoutProxy('foo', timeout=1.)
 
 
+def test_agent_proxy_wait_running(nsproxy):
+    """
+    Using `wait_for_running` on a proxy after initialization should block until
+    the agent is running or time out.
+    """
+    ap = AgentProcess('agent')
+    ap.start()
+
+    # Get "offline" proxy
+    agent = Proxy('agent')
+    Timer(.5, agent.run).start()
+
+    # Wait long enough for the agent to start running
+    proxy = Proxy('agent', timeout=1).wait_for_running()
+    assert proxy.ping() == 'pong'
+
+
+def test_agent_proxy_wait_running_timeout(nsproxy):
```

Comment:
This test and the last one share the same code. You should merge them into one that simply tests the TimeoutError exception when the agent is not running.

Use `@pytest.mark.parametrize` to change the timeout value (you can find a couple of examples in the osBrain test suite too).

Reply:
I didn't even know this was a possibility, but looking at the other tests, I can see it makes a lot of sense.

Reply:
I guessed so. `pytest` is a great tool. :blush:

Reply:
Definitely. I'll have to investigate what other things can it do